src/graphql/root.rs

```diff
@@ -395,7 +395,14 @@ graphql_object!(Query: Context |&self| {
 
     /// Find all the message types we know about
     field messageTypes(&executor) -> FieldResult<Vec<String>>{
-        Err("Not implemented")?
+        let connection = executor.context().connection.lock()?;
+        let query = messages_table
+            .select(messages_content_type)
+            .filter(messages_content_type.is_not_null())
+            .distinct()
+            .load::<Option<String>>(&(*connection))?;
+        let result : Vec<String> = query.into_iter().filter(|x| x.is_some()).map(|x| x.unwrap() ).collect();
```

Comment:
This line is essentially taken from here: http://databasefaq.com/index.php/answer/48568/iterator-rust-optional-convert-list-of-optiont-to-list-of-t-when-t-cannot-be-copied

Reply:
We can tidy this up just a little using [filter_map](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.filter_map):

```rust
/// Find all the message types we know about
field messageTypes(&executor) -> FieldResult<Vec<String>>{
    let connection = executor.context().connection.lock()?;
    let result = messages_table
        .select(messages_content_type)
        .filter(messages_content_type.is_not_null())
        .distinct()
        .load::<Option<String>>(&(*connection))?
        .into_iter()
        .filter_map(|message_type|{message_type});

    Ok(result)
}
```

Reply:
I'm keen to merge this though. If you've got energy to add that filter_map change, go for it and push it up.

If not, let me know and I'll merge it as is.